pkg/server/cloud/aws/instances.go

```diff
@@ -231,6 +232,25 @@ func bootImageSpecToDescribeImagesInput(spec cloud.BootImageSpec) *ec2.DescribeI
 	return input
 }
 
+func getRootDeviceName(img *ec2.Image) (string, error) {
+	if img.RootDeviceName == nil {
+		msg := "cannot get root device name from image: " + *img.Name
+		return "", errors.New(msg)
+	}
+	return *img.RootDeviceName, nil
+}
+
+func getRootDeviceVolumeSize(blockDevices []*ec2.BlockDeviceMapping, rootDeviceName string) *int32 {
+	var rootDiskSize int32
+	for _, blockDevice := range blockDevices {
+		if *blockDevice.DeviceName == rootDeviceName {
+			rootDiskSize = int32(*blockDevice.Ebs.VolumeSize)
```

Comment:
Not sure if these pointers are ever `nil`, but I'd also use the `aws.*Value()` helpers here.

Reply:
To be a bit clearer: The main issue here is that `*blockDevice.Ebs.VolumeSize` dereferences `VolumeSize`. If that value is `nil`, this will panic.  Below there's a dance to take the address of `rootDiskSize` and pass that through `aws.Int32Value`.  That's not needed as `rootDiskSize` is already an int32 so that's basically a strange noop with the overhead of a function call.